src/controllers/queries/index.ts

```diff
@@ -511,26 +465,81 @@ export class QueriesController extends HTMLController {
     )
   }
 
-  private async submitDrpcRequest({
-    queryId,
+  private async submitQueryResponse({
+    connection,
+    queryRow,
+    localQuery,
+    log,
+    rpcResponse = undefined,
+  }: {
+    log: Logger
+    connection: ConnectionRow
+    queryRow: QueryRow
+    rpcResponse?: DrpcResponse | undefined
+    localQuery: { emissions: string; queryIdForResponse: UUID }
+  }): Promise<HTML> {
+    if (!connection.agent_connection_id) return this.handleError(log, queryRow, connection)
+    try {
+      rpcResponse = await this.cloudagent.submitDrpcRequest(connection.agent_connection_id, 'submit_query_response', {
+        query: 'Scope 3 Carbon Consumption',
+        ...localQuery,
+      })
+      log.info('submitting DRPC request %j', { rpcResponse, localQuery })
+      if (!rpcResponse) {
+        return this.handleError(log, queryRow, connection)
+      }
+    } catch (err) {
+      log.warn('DRPC has failed %j', err)
+      return this.handleError(log, queryRow, connection, undefined, err)
+    }
+    const { result, error, id: rpcId } = rpcResponse
+
+    log.info('persisting query_rpc response', rpcResponse)
+    await this.db.insert('query_rpc', {
+      agent_rpc_id: rpcId,
+      query_id: queryRow.id,
+      role: 'client', // am I still a client when I'm a 'responder'?
+      method: 'submit_query_request',
+      result,
+      error,
+    })
+
+    if (!result || error) {
+      log.warn('error happened while persisting query_rpc %j', error)
+      return this.handleError(log, queryRow, connection, rpcId)
+    }
+    await this.db.update('query', { id: queryRow.id }, { query_response: localQuery.emissions, status: 'resolved' })
+
+    return this.html(
+      this.scope3CarbonConsumptionResponseTemplates.newScope3CarbonConsumptionResponseFormPage({
+        formStage: 'success',
+        company: connection,
+        query: queryRow,
+      })
+    )
+  }
+
+  private async submitQueryRequest({
```

Comment:
I think we can still deduplicate this functionality from `scope3CarbonConsumptionStage` as per the previous PR's comments

Reply:
This has been updated, so know we have single function (yay!) `submitDrpcRequest()` that works for partials/stage/regular